hogql_parser/parser.cpp

```diff
@@ -1819,6 +1819,47 @@ class HogQLParseTreeConverter : public HogQLParserBaseVisitor {
     RETURN_NEW_AST_NODE("ArrayAccess", "{s:N,s:N}", "array", object, "property", property);
   }
 
+  VISIT(ColumnExprNullArrayAccess) {
+    PyObject* property = visitAsPyObject(ctx->columnExpr(1));
+    int is_property_a_constant = is_ast_node_instance(property, "Constant");
+    if (is_property_a_constant == -1) {
+      Py_DECREF(property);
+      throw PyInternalError();
+    }
+    if (is_property_a_constant) {
+      PyObject* property_value = PyObject_GetAttrString(property, "value");
+      if (!property_value) {
+        Py_DECREF(property);
+        throw PyInternalError();
+      }
+      PyObject* zero = PyLong_FromLong(0);
+      if (!zero) {
+        Py_DECREF(property_value);
+        Py_DECREF(property);
+        throw PyInternalError();
+      }
+      int is_property_zero = PyObject_RichCompareBool(property_value, zero, Py_EQ);
+      Py_DECREF(zero);
+      Py_DECREF(property_value);
+      if (is_property_zero == -1) {
+        Py_DECREF(property);
+        throw PyInternalError();
+      }
+      if (is_property_zero) {
+        Py_DECREF(property);
+        throw SyntaxError("SQL indexes start from one, not from zero. E.g: array[1]");
+      }
+    }
+    PyObject* object;
+    try {
+      object = visitAsPyObject(ctx->columnExpr(0));
+    } catch (...) {
+      Py_DECREF(property);
+      throw;
+    }
+    RETURN_NEW_AST_NODE("ArrayAccess", "{s:N,s:N,s:O}", "array", object, "property", property, "nullish", Py_True);
```

Comment:
All changes in this file are exact copies of the non-`Null` variants (`ColumnExprNullArrayAccess` -> `ColumnExprArrayAccess`) with `nullish: True` added here.